### PR TITLE
add pokemon api

### PR DESCRIPTION
**Pull request by vherus**

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -22,6 +23,7 @@ app.get('/', (_req: Request, res: Response) => {
 });
 
 app.use('/:username/todo', Username, todoRouter);
+app.use('/:username/pokemon', Username, pokemonRouter);
```
Comment (Luca-Terrazzan) [inline src/server.ts:26]: Why do we need the github username to use pokemon CRUD?

Comment (vherus) [reply src/server.ts:26]: Because each student will be modifying their own data set

Comment (Luca-Terrazzan) [reply src/server.ts:26]: Makes sense

File: docs/api.yml [hunk 2]
```diff
@@ -255,6 +257,178 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BadRequest'
+        '401':
+          description: The username provided is not part of the boolean-uk github organisation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GithubAuthIssue'
+    put:
```
Comment (Luca-Terrazzan) [inline docs/api.yml:338]:
```suggestion
  /{username}/pokemon/{pokemonId}:
    put:
```

Comment (Luca-Terrazzan) [reply docs/api.yml:338]: This should put the pokemon id in the URL properly while using swagger

Comment (vherus) [reply docs/api.yml:338]: Noice, fixed

**Review by Luca-Terrazzan — APPROVED** [review]

**Review by Luca-Terrazzan — APPROVED** [review]

**Pull request merged**